qml/MainPage.qml

```diff
@@ -13,38 +13,104 @@ Page {
             return new RegExp('');
         }
     }
+    property bool searchMode: false
```

Comment:
You can use the `state` property to switch to the search state (the state needs to have a name) avoiding this property (and if you need any comparisons, you can do `mainPage.state === 'SEARCH'`)